blockmanager.go

```diff
@@ -1084,14 +1116,15 @@ out:
 			switch msg := m.(type) {
 			case *newPeerMsg:
 				b.handleNewPeerMsg(candidatePeers, msg.peer)
+				b.PeerProcessed()
 
 			case *txMsg:
 				b.handleTxMsg(msg)
-				msg.peer.txProcessed <- struct{}{}
+				b.TxProcessed()
```

Comment:
This needs to be a channel in the peer data instead of the block manager as it's unique to the peer that is processing it.  As you've changed it currently, you've made it so getting a tx or block from *any* peer will cause the code that is blocking to erroneously continue.